rest/config.go

```diff
@@ -989,9 +996,44 @@ func ValidateConfigOrPanic(runMode SyncGatewayRunMode) {
 
 }
 
+func RegisterSignalHandler() {
+	signalchannel := make(chan os.Signal, 1)
+	signal.Notify(signalchannel, syscall.SIGHUP, os.Interrupt, os.Kill)
+
+	go func() {
+		for sig := range signalchannel {
+			base.Infof(base.KeyAll, "Handling signal: %v", sig)
+			switch sig {
+			case syscall.SIGHUP:
+				HandleSighup()
+			case os.Interrupt, os.Kill:
+				// We'll add a small delay here for graceful shutdown so that any
+				// log collation buffers have time to be flushed to the outputs.
+				time.Sleep(logCollationBufferFlushDelay)
+				os.Exit(130) // 130 == exit code 128 + 2 (interrupt)
+			}
+		}
+	}()
+}
+
+func panicHandler() (panicHandler func()) {
+	return func() {
+		// Delay any panics to allow log collation buffers to flush to the outputs.
+		if r := recover(); r != nil {
+			base.Errorf(base.KeyAll, "Handling panic: %v", r)
+			time.Sleep(logCollationBufferFlushDelay)
```

Comment:
Could switch this to a call like FlushLogs() that lives with the rest of the logging code, to keep the functionality clear.  If it makes sense to keep it here, move the comment down.